src/action/applyMigrations.ts

```diff
@@ -58,29 +58,41 @@ export default async function ({ environment, defaultLocale }: { environment: En
   };
 
   Logger.verbose('Run migrations and update version entry');
-  // Allow mutations
-  let migrationToRun;
-  let lastMigration;
+
+  let lastMigration: string | undefined;
 
   /* eslint-disable no-await-in-loop */
-  while ((migrationToRun = migrationsToRun.shift())) {
-    const filePath = path.join(MIGRATIONS_DIR, versionToFilename(migrationToRun));
-    Logger.verbose(`Running ${filePath}`);
-    await runMigration(
-      Object.assign(migrationOptions, {
+  try {
+    for (const migrationToRun of migrationsToRun) {
+      const filePath = path.join(MIGRATIONS_DIR, versionToFilename(migrationToRun));
+      Logger.verbose(`Running ${filePath}`);
+
+      await runMigration({
+        ...migrationOptions,
         filePath,
-      }),
+      });
+
+      lastMigration = migrationToRun;
+      Logger.success(`Migration script ${migrationToRun}.js succeeded`);
+    }
+  } catch (error) {
+    const brokenMigrationIdx = Math.max(migrationsToRun.indexOf(lastMigration ?? '') - 1, 0);
```

Comment:
the error can only be thrown from `await runMigration(...)`, so `lastMigration` will not be updated to the current `migrationToRun` value. should we be adding 1 instead of subtracting 1 to the indexOf value?

Reply:
nice catch @waisangu. Or what about just wrapping the 
`await runMigration({
        ...migrationOptions,
        filePath,
      });`

in a try/catch. I believe it would make things simpler.

Reply:
nice catch, ty